cmd/capibmadm/cmd/powervs/image/import.go

```diff
@@ -0,0 +1,209 @@
+limitations under the License.
+*/
+
+package image
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	powerClient "github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/powervs"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+)
+
+type imageImportOptions struct {
+	BucketName    string
+	Region        string
+	ImageFilename string
+	ImageName     string
+	AccessKey     string
+	SecretKey     string
+	StorageType   string
+	InstanceID    string
+	Public        bool
+	Watch         bool
+	WatchTimeout  time.Duration
+}
+
+// ImportCommand import image PowerVS instances.
+func ImportCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "import",
+		Short: "Import the image into PowerVS instances",
+		Example: `
+# Import image in PowerVS instances.
+export IBMCLOUD_API_KEY=<api-key>
+# import image using default storage type (service credential will be autogenerated)
+capibmadm powervs image import --service-instance-id <service-instance-id> -b <bucketname> --object rhel-83-10032020.ova.gz --pvs-image-name test-image -r <region> --zone <zone>
+
+# import image using default storage type with specifying the accesskey and secretkey explicitly
+capibmadm powervs image import --service-instance-id <service-instance-id> -b <bucketname> --accesskey <accesskey> --secretkey <secretkey> --object rhel-83-10032020.ova.gz --pvs-image-name test-image -r <region> --zone <zone>
+
+# with user provided storage type
+capibmadm powervs image import --service-instance-id <service-instance-id> -b <bucketname> --pvs-storagetype <storagetype> --object rhel-83-10032020.ova.gz --pvs-image-name test-image -r <region> --zone <zone>
+
+# import image from a public IBM Cloud Storage bucket
+capibmadm powervs image import --service-instance-id <service-instance-id> -b <bucketname> --zone <zone> -r <region> --object rhel-83-10032020.ova.gz --pvs-image-name test_image --public-bucket 
+`,
+	}
+	var imageImportOption imageImportOptions
+	cmd.Flags().StringVarP(&imageImportOption.BucketName, "bucket", "b", "", "Cloud Object Storage bucket name.")
+	cmd.Flags().StringVarP(&imageImportOption.Region, "bucket-region", "r", "", "Cloud Object Storage bucket location.")
+	cmd.Flags().StringVar(&imageImportOption.ImageFilename, "object", "", "Cloud Object Storage object name.")
+	cmd.Flags().StringVar(&imageImportOption.AccessKey, "accesskey", "", "Cloud Object Storage HMAC access key.")
+	cmd.Flags().StringVar(&imageImportOption.SecretKey, "secretkey", "", "Cloud Object Storage HMAC secret key.")
+	cmd.Flags().StringVar(&imageImportOption.ImageName, "pvs-image-name", "", "Name to PowerVS imported image.")
```

Comment:
wondering if we can rename this flag just `--name` because pvs, image already present in the command name

Reply:
Done. Renamed it to `--name`.